app/controllers/sessions/edit_controller.rb

```diff
@@ -3,6 +3,18 @@
 class Sessions::EditController < ApplicationController
   before_action :set_session
 
+  def edit_programmes
+    render :programmes
+  end
+
+  def update_programmes
+    if @session.update(programmes_params)
```

Comment:
Should we be checking `@session.can_change_programmes?` here too?

Reply:
Yeah good point, will amend.

Reply:
This has been fixed now.